x-pack/plugins/secops/public/components/timeline/body/events/index.tsx

```diff
@@ -0,0 +1,177 @@
+                          data-test-subj="actions"
+                          eventId={event._id!}
+                          eventIsPinned={eventIsPinned({
+                            eventId: event._id!,
+                            pinnedEventIds,
+                          })}
+                          getNotesByIds={getNotesByIds}
+                          noteIds={eventIdToNoteIds[event._id!] || emptyNotes}
+                          onEventToggled={this.onToggleExpanded(event._id!)}
+                          onPinClicked={getPinOnClick({
+                            allowUnpinning: !eventHasNotes(eventIdToNoteIds[event._id!]),
+                            eventId: event._id!,
+                            onPinEvent,
+                            onUnPinEvent,
+                            pinnedEventIds,
+                          })}
+                          showNotes={!!this.state.showNotes[event._id!]}
+                          toggleShowNotes={this.onToggleShowNotes(event._id!)}
+                          updateNote={updateNote}
+                        />
+                      </EuiFlexItem>
+
+                      <EuiFlexItem grow={true}>
+                        <Columns
+                          data-test-subj="columns"
+                          columnHeaders={columnHeaders}
+                          columnRenderers={columnRenderers}
+                          ecs={event}
+                        />
+                        <NoteCards
+                          associateNote={this.associateNote(event._id!, addNoteToEvent, onPinEvent)}
+                          data-test-subj="note-cards"
+                          getNewNoteId={getNewNoteId}
+                          getNotesByIds={getNotesByIds}
+                          noteIds={eventIdToNoteIds[event._id!] || emptyNotes}
+                          showAddNote={!!this.state.showNotes[event._id!]}
+                          toggleShowAddNote={this.onToggleShowNotes(event._id!)}
+                          updateNote={updateNote}
+                          width="90%"
+                        />
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  </EuiFlexItem>
+
+                  <EuiFlexItem data-test-subj="event-details" grow={true}>
+                    <ExpandableEvent
+                      event={event}
+                      forceExpand={!!this.state.expanded[event._id!]}
+                      hideExpandButton={true}
+                      stringifiedEvent={stringifyEvent(event)}
+                      timelineId={id}
+                    />
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              </>
+            )}
+          </EuiFlexItem>
+        ))}
+      </EuiFlexGroup>
+    );
+  }
+
+  private onToggleShowNotes = (eventId: string): (() => void) => () => {
+    this.setState(state => ({
+      showNotes: {
+        ...state.showNotes,
+        [eventId]: !!!state.showNotes[eventId],
```

Comment:
Did you want 3 ❗️ ❗️ ❗️ marks and not two or maybe one? Even if that's `null` or `undefined` it will return true with just one `!` point.

```
There is no difference between !a and !!!a, since !!!a is just !!(!a) and
because !a is a boolean, !!(!a) is just its double negation, therefore the same.
```

Ref: https://stackoverflow.com/questions/21154510/the-use-of-the-triple-exclamation-mark

Let me know if I'm missing something. I think I'm sensitive to the amount of exclamations right now ;-)